src/core/net/ip6_address.cpp

```diff
@@ -249,6 +249,11 @@ void Address::SetToRealmLocalAllMplForwarders(void)
     *this = GetRealmLocalAllMplForwarders();
 }
 
+void Address::SetPrefix(const NetworkPrefix &aNetworkPrefix)
+{
+    mFields.mComponents.mNetworkPrefix = aNetworkPrefix;
+}
+
 void Address::SetPrefix(const uint8_t *aPrefix, uint8_t aPrefixLength)
```

Comment:
Just testing our method name policy here: to overload or not? Thoughts?

Reply:
I initially had  this as `SetNetworkPrefix()` but then decided in favor of overloading the `SetPrefix()` method.
- I think the overloads of `SetPrefix()` are distinct (and easily distinguishable) on the caller side, one requires a single reference argument, and the other one has two arguments, a buffer pointer and a length.
- Also in both cases, we don't do any conversion and directly copy/use the provided input value.

In future we may add a new overload of the form `SetPrefix(const otIp6Prefix &)`. I feel this will still be ok and distinct (and again value would be copied as is - no conversion).

Thoughts?

Reply:
I'm personally fine with the method overloading here. I'm just trying to reconcile this with what you stated in https://github.com/openthread/openthread/pull/5194#discussion_r450412604 :

> - For cases where the value is directly used (no change/conversion), either way (use explicit method name or overload) is fine.
>   - I personally prefer explicit name in this case as well (feel it is more readable and avoid any incorrect use), but we can leave it to developer.

Just want to ensure consistency.

Reply:
Very good and fair point indeed. 😄 

Thinking more about this (and actually somewhat changing my mind on this), here is what I would suggest.
- For cases where there is a conversion/parsing happening we prefer/recommend using explicit method name vs overloading (to make it super clear that it being converted).
- For cases where there is no conversion (value is used directly), we prefer/recommend using overloading over explicit names.
- In both case, it's a recommendation (not a requirement) so we keep it flexible for handling special situations.

Thoughts?

Reply:
Sounds good to me!